docs/contributing/organization/release-process.rst

```diff
@@ -0,0 +1,73 @@
+The Release Process
+===================
+
+This document explains the **release process** of the Sylius project (i.e. the
+code & documentation hosted on the main ``Sylius/Sylius`` `Git repository`_).
+
+Sylius manages its releases through a *time-based model* and follows the
+`Semantic Versioning`_ strategy:
+
+* A new Sylius minor version (e.g. 1.1, 2.4, 3.2) comes out every *six months*:
+* A new Sylius major version (e.g., 2.0, 3.0) comes out every *two years* and
+  it's released at the same time of the last minor version of the previous major
+  version.
+
```

Comment:
Can also be something said about patches? I'm now following dev-master in my current project, which I update every 2-3 weeks. I would like to stay up to date with the project on stable patches with easy upgrades. Upgrading every 6 months once, means an huge changelog and potentially lots of updates to your own customisations. The less big the updates are or the more options you have to keep up to date, the better in my opinion. Nearly no project has every customisation covered with behat scenarios, unit / integration tests, so you need to read all the changes anyway, so if that's a small diff it's relatively easy.